### ISSUE TITLE
ActiveSupport Samoa Timezone error

### ISSUE THREAD

**Issue by jacquesporveau** [rails/rails#27419]
### Steps to reproduce

In rails console.

`DateTime.now.in_time_zone("Samoa")` Returns  a date with a UTC offset of + 14:00
`ActiveSupport::TimeZone.new('Samoa').utc_offset/60/60` Returns -11

### Expected behavior

`ActiveSupport::TimeZone.new('Samoa').utc_offset/60/60` Should return 14 or 13 based on daylight savings time

### Actual behavior

`ActiveSupport::TimeZone.new('Samoa').utc_offset/60/60` Returns -11

### System configuration
**Rails version**: 5.0.0.1

**Ruby version**: 2.3.0p0

**Comment by garygao** [comment]
@jacquesporveau 
I think this is an issue of the gem "tzinfo", run the following code in irb, you also will get -11:
```
2.3.0 :001 > require 'tzinfo'
 => true
2.3.0 :001 > TZInfo::Timezone.get('Pacific/Apia').current_period.utc_offset/60/60
 => -11
```
**tzinfo version**: 1.2.2

**Comment by philr** [comment]
This issue has been fixed in tzinfo [version 1.2.3](https://github.com/tzinfo/tzinfo/releases/tag/v1.2.3).

See tzinfo/tzinfo#66 for further details.

Please note that the expected behaviour of `ActiveSupport::TimeZone.new('Samoa').utc_offset/60/60` is to always return 13 regardless of whether daylight savings time is being observed (the `utc_offset` method gives the base offset of the time zone).
